src/ui/model.rs

```diff
@@ -0,0 +1,129 @@
+use crate::core::{dir_stats::DirStats, MatchData};
+use ratatui::widgets::TableState;
+use size::Size;
+use std::path::PathBuf;
+
+#[derive(Debug, Clone)]
+pub struct MatchDataUI {
+    pub idx: usize,
+    pub data: MatchData,
+    pub dir_stats: DirStats,
+    pub status: MatchDataUIStatus,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum MatchDataUIStatus {
+    Found,
+    Selected,
+}
+
+#[derive(Debug, Clone)]
+pub struct TableData {
+    pub state: TableState,
+    pub data: Vec<MatchDataUI>,
+    cleanable_space: Size,
+}
+
+impl Default for TableData {
+    fn default() -> Self {
+        Self {
+            state: Default::default(),
+            data: Default::default(),
+            cleanable_space: Size::from_bytes(0),
+        }
+    }
+}
+
+impl TableData {
+    pub fn add_match(&mut self, data: MatchData) {
+        let idx = self.data.len();
+        self.data.push(MatchDataUI {
+            idx,
+            data,
+            dir_stats: DirStats::default(),
+            status: MatchDataUIStatus::Found,
+        });
+        self.resort();
+    }
+
+    pub fn update_match(&mut self, idx: usize, data: DirStats) -> bool {
+        if let Some(ele) = self.get_by_idx_mut(idx) {
+            ele.dir_stats = data;
+            if let Some(size) = data.size {
+                self.cleanable_space += size;
+                return true;
+            }
+        }
+        false
+    }
+
+    pub fn resort(&mut self) {
```

Comment:
Maybe rename that function to sort_again or something like that because I was confused by the current name for a moment when reading this code at first